pyproject.toml

```diff
@@ -30,7 +30,7 @@ dependencies = [
     "datasets~=2.14.6,<3.0.0",
     "requests",
     "optuna",
-    "pydantic==2.5.0",
+    "pydantic~=2.5.0,<2.5.3",
```

Comment:
ideally, we don't want to update the dependencies if not needed, but tagging @okhat for context here

Reply:
okay, waiting for  @okhat comment, since it has a lot of implications about my current implementation. I was wondering why a strict requirement on pydantic if overall moving from 2.5.0 to 2.5.3 doesn't affect the package that much.

Reply:
No need, I removed the pydantic version change